staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -851,6 +1003,70 @@ func (alloc *allocator) createNodeSelector(result []internalDeviceResult) (*v1.N
 	return nil, nil
 }
 
+type requestAccessor struct {
```

Comment:
Can we make this an interface with two different implementations, one for `type deviceRequestAccessor *resourceapi.DeviceRequest` and another for `type deviceSubRequestAccessor *resourceapi.DeviceSubRequest`? Then all of the if checks go away.